pkg/api/configsync/v1beta1/resource_override.go

```diff
@@ -93,29 +93,36 @@ type RootSyncOverrideSpec struct {
 	// +optional
 	NamespaceStrategy configsync.NamespaceStrategy `json:"namespaceStrategy,omitempty"`
 
-	// clusterRole controls which role to bind the service account for this RootSync's
-	// reconciler to.
+	// roleRefs is a list of Roles or ClusterRoles to create bindings.
+	// If unset, a binding to cluster-admin will be created.
 	//
 	// +optional
-	ClusterRole ClusterRoleOverrideSpec `json:"clusterRole,omitempty"`
+	RoleRefs []RootSyncRoleRef `json:"roleRefs,omitempty"`
 }
 
-// ClusterRoleOverrideSpec allows to override settings for the root reconciler permissions
-type ClusterRoleOverrideSpec struct {
-	// name is the name of the ClusterRole to bind to; defaults to "cluster-admin".
-	//
-	// If you specify a different name, the ClusterRole needs to be created manually.
+// each item references a Role or ClusterRole to create
+// a binding to for this reconciler. It supports a namespace field that can be used
+// to create RoleBindings rather than ClusterRoleBindings.
+//
+//nolint:revive
+type RootSyncRoleRef struct {
+
+	// kind refers to the Kind of the RBAC resource.
+	// Accepted values are Role and ClusterRole.
 	//
-	// +optional
+	// +kubebuilder:validation:Pattern=^(Role|ClusterRole)$
+	Kind string `json:"kind,omitempty"`
```

Comment:
Are these all required when you don't specify +optional?
If it's required, should it still have omitempty?

Reply:
Same for Name

Reply:
You're right, the `omitempty` parameter effectively made these optional. I removed `omitempty`, and they should be properly validated as required by the CRD now.